src/uct/tcp/tcp_iface.c

```diff
@@ -276,20 +279,23 @@ static UCS_CLASS_INIT_FUNC(uct_tcp_iface_t, uct_md_h md, uct_worker_h worker,
 
     UCS_CLASS_CALL_SUPER_INIT(uct_base_iface_t, &uct_tcp_iface_ops, md, worker,
                               params, tl_config
-                              UCS_STATS_ARG((params->field_mask & 
+                              UCS_STATS_ARG((params->field_mask &
                                              UCT_IFACE_PARAM_FIELD_STATS_ROOT) ?
                                             params->stats_root : NULL)
                               UCS_STATS_ARG(params->mode.device.dev_name));
 
     ucs_strncpy_zero(self->if_name, params->mode.device.dev_name,
                      sizeof(self->if_name));
-    self->outstanding            = 0;
-    self->config.buf_size        = config->super.max_bcopy +
-                                   sizeof(uct_tcp_am_hdr_t);
-    self->config.prefer_default  = config->prefer_default;
-    self->config.max_poll        = config->max_poll;
-    self->sockopt.nodelay        = config->sockopt_nodelay;
-    self->sockopt.sndbuf         = config->sockopt_sndbuf;
+    self->outstanding           = 0;
+    self->config.buf_size       = config->super.max_bcopy +
+                                  sizeof(uct_tcp_am_hdr_t);
+    self->config.short_size     = ucs_max(config->super.max_short,
```

Comment:
does this mean that MAX_SHORT var will not work properly for TCP? Maybe it's better to set bigger MAX_SHORT by default for TCP? (can be even equal to max_bcopy).

Reply:
done by overriding `UCX_TCP_MAX_SHORT` env